File: clientlibs/js/src/functions/setWorkingGroup.ts
import { UpGradeClientInterfaces, UpGradeClientEnums } from '../types';
import fetchDataService from '../common/fetchDataService';

export default async function setWorkingGroup(
  customHttpClient: UpGradeClientInterfaces.ICustomHttpClient,
  url: string,
  userId: string,
  token: string,
  clientSessionId: string,
  workingGroup: Record<string, string>
): Promise<UpGradeClientInterfaces.IUser> {
  const response = await fetchDataService(
    customHttpClient,
    url,
    token,
    clientSessionId,
    { id: userId, workingGroup: workingGroup },
    UpGradeClientEnums.REQUEST_TYPES.PATCH
  );
  if (response.status) {
    return response.data;
  } else {
    throw new Error(JSON.stringify(response.message));
  }
}

File: clientlibs/csharp/src/functions/SetWorkingGroup.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using Enums;
using Interfaces;
using Newtonsoft.Json;

public static class WorkingGroupClass
{
  public static async Task<WorkingGroup> SetWorkingGroup(
    string url,
    string userId,
    string token,
    string clientSessionId,
    Dictionary<string, string> workingGroup)
  {
    var response = await FetchDataServiceClass.FetchDataService(
        url,
        token,
        clientSessionId,
        new { id = userId, workingGroup },
        REQUEST_TYPES.PATCH
    );
    if (response.status)
    {
      var responseData = JsonConvert.DeserializeObject<WorkingGroup>(response.data.ToString());
      return responseData as WorkingGroup;
    }
    else
    {
      throw new Exception(JsonConvert.SerializeObject(response.message));
    }
  }
}

